trigger.js

```diff
@@ -206,6 +215,44 @@ class TriggerHappy {
         Hooks.once('updateToken', () => this._executeTriggers(triggers));
         return true;
     }
+  
+    _onPreUpdateToken(scene, id, update) {
+        if (!scene.isView) return true;
+        if (update.x === undefined && update.y === undefined) return true;
+        const token = scene.data.tokens.find(t => t._id === update._id);
+        if (token.hidden) return true; // don't stop ivnisible tokens?
+
+        // Get all trigger tokens in scene
+        let targets = this._getTriggerTokens(canvas.tokens.placeables, this.triggers, 'capture');
+        if (!targets) return true;
+
+        let x = update.x || token.x;
+        let  y = update.y || token.y;
+        let tw = token.width * canvas.scene.data.grid / 2;
+        let th = token.height * canvas.scene.data.grid / 2;
+        let motion = new Ray({x: token.x + tw, y: token.y + th}, {x: x + tw, y: y + th});
+
+        // don't trigger on tokens that are already captured
+        targets = targets.filter(target => token.x !== target.x + target.w  / 2 - tw || token.y !== target.y + target.h / 2 - th)
+        
+        // sort list by distance from start token position
+        targets.sort((a , b) => dist(token.x, token.y, b.x, b.y) - dist(token.x, token.y, a.x, a.y));

```

Comment:
foundry.js uses `Math.hypot` to calculate distances between coordinates. 
Also for the arguments, `(a, b)` instead of `(a , b)` (extra space after `a`).
How about this ? 
`targets.sort((a, b) => Math.hypot(token.x - b.x, token.y - b.y) - Math.hypot(token.x - a.x, token.y - a.y))`

Reply:
It was an efficiency thing, Math.hypot will do a square root to get the actual value. If the compiler is good not doing the square root is faster, but if not Math.hypot will be faster. I don't feel strongly about it.

Reply:
I would assume on the kind of machines (and browser versions) needed to run Foundry that Math.hypot is going to be a single cpu instruction. I suggested it more about the ease of reading and the fact that that's what foundry does to calculate distances rather than about optimization. I don't feel strongly about it either, but it does look nicer/easier to read I think with Math.hypot, and I guess more optimized...